client/src/components/mapview/helpers.js

```diff
@@ -20,8 +20,30 @@ const generateTimes = () => {
   }
 };
 
-const convertToUNIXTime = (startTime, endTime, startDate, endDate) => {
-  console.log('these values will be converted to UNIX time: ', startTime, endTime, startDate, endDate);
+const parseTime = (timeString) => { // convert the input time string to an hour number on a 24-hour scale
+  let hrs = Number(timeString.split(':')[0]); // get hours (reservations are made on the hour only)
+  if (hrs === 12 && timeString.indexOf('AM') > -1) { // edge case: it's 12:00AM --> return 0
+      return 0;
+  }
+  if (timeString.indexOf('PM') > - 1 && hrs < 12) { // if it's PM and after 12, add 12 to the hour to convert to 24 hr timescale
+    hrs+=12;
+  }
+  return hrs;
+};
+
+const convertToUNIXTime = (time, date) => {
+  const hour = parseTime(time);
+  const year = date.getFullYear();
+  const month = date.getMonth();
+  const day = date.getDate();
+  // date.setHours(hour); // use these lines if you don't want to create a new time object for some reason
+  // date.setMinutes(0);
+  // date.setSeconds(0);
+  // date.setMilliseconds(0);
+  // console.log('result after update to the existing date object: ', date.getTime());
+  console.log('human readable reservation time: ', new Date(year, month, day, hour));
+  console.log('UNIX time of reservation: ', new Date(year, month, day, hour).getTime());
+  return new Date(year, month, day, hour).getTime();
```

Comment:
My two cents:
1. My understanding is Unix Time is currently 10 digits, and shall be "new Date().getTime / 1000". Without that "/1000" it's in in milliseconds. (However, this will have to be same to booking section. I'm not sure which one will be used for booking and I just added this comment based on your var/log name.)

2. getMonth() returns month from 0-11 not 1-12.
The way you set up the month and new Date() works since it will cancel out. However, shall we let month represent the actual month, not month - 1? Like below:
"
const month = date.getMonth() + 1;
// console.log('date params: ', year, month, day, hour);   // could log month to confirm
return new Date(year, month - 1, day, hour).getTime()/1000;    // also -1 for your two logs
"

Reply:
Resolved in latest PR.